edge-agent/src/Microsoft.Azure.Devices.Edge.Agent.Core/logs/ILogsProcessor.cs

```diff
@@ -0,0 +1,14 @@
+// Copyright (c) Microsoft. All rights reserved.
+namespace Microsoft.Azure.Devices.Edge.Agent.Core.Logs
+{
+    using System.Collections.Generic;
+    using System.IO;
+    using System.Threading.Tasks;
+
+    public interface ILogsProcessor
```

Comment:
should it be ILogProcessor?

Reply:
Emm.. it processes the logs, so I called it LogsProcessor.. 
both seem equally odd to me :)

Reply:
The idea here is that the `Provider` gives chunks of bytes and the `Processor` returns lists of log messages?

Reply:
Provider is the higher level class - given a settings/config object, it gets the right stream, processes it (using LogsProcessor, Compressor, etc), and returns it. 
LogsProcessor uses akka streams to process the incoming stream (de-frame it, parse it, etc)